File: CSA2/sourse files/plant.cpp
//------------------------------------------------------------------------------
// plant.cpp - содержит реализацию класса растения
//------------------------------------------------------------------------------

#include "plant.h"

//------------------------------------------------------------------------------
// Конструктор для создания растения по имени
Plant::Plant(std::string name) {
    Name = name;
}

//------------------------------------------------------------------------------
// Получение частного от деления числа гласных букв в названии на общую длину названия
double Plant::GetPartOfVowels() {
    int i, vow = 0;

    for (i = 0; Name[i]; i++)
    {
        if (Name[i] == 'a' || Name[i] == 'e' || Name[i] == 'i' || Name[i] == 'o' || Name[i] == 'u'
            || Name[i] == 'A' || Name[i] == 'E' || Name[i] == 'I' || Name[i] == 'O' || Name[i] == 'U')
        {
            vow++;
        }
    }
    return ((double)vow) / i;
}

File: CSA3/sourse files/plant.py
# ----------------------------------------------
class Plant:
    def __init__(self, name):
        self.name = name

    # Возвращает строковое представление растения
    def toString(self):
        pass

    # Посчитать количество гласных в названии
    def getPartOfVowels(self):
        lowercase = self.name.lower()
        vowel_counts = {}
        for vowel in "aeiou":
            count = lowercase.count(vowel)
            vowel_counts[vowel] = count
        return sum(vowel_counts.values()) / len(self.name)
